### DISCUSSION TITLE
navbar  in hiddenBreakpoint={'xl'}  and hide, but rightContent not grow full width

### DISCUSSION THREAD

**Discussion by vaynevayne** [mantinedev/mantine#4507] [Q&A]
How can I hide the navbar when the right inside automatically fills the entire width? 

```jsx
 <AppShell
        padding="md"
 
        navbar={<Navbar authRouters={authRouters} hiddenBreakpoint={'xl'}   width={{ sm: 250 }}
      hidden={!leftOpened}
      p="md"
       />}
        header={<MyHeader authRouters={authRouters} />}
        styles={(theme) => ({
          main: {
            backgroundColor:
              theme.colorScheme === 'dark' ? theme.colors.dark[8] : theme.colors.gray[0],
          },
        })}
      >
        <MyErrorBoundary>
       
          <Suspense fallback={<Loader />}>
            <Outlet context={{ routers: authRouters }} />
          </Suspense>
        </MyErrorBoundary>
      </AppShell>
```

**Comment by vaynevayne** [comment]
@rtivital hello, Could you look at it for me, please

**Reply by rtivital** [reply]
I do not understand the question

**Comment by vaynevayne** [comment]
<img width="1171" alt="image" src="https://github.com/mantinedev/mantine/assets/124666577/dcf7d3e8-97b4-48b9-9df2-f0c9fe308955">



@rtivital  Look at this code, `1024 < xl`, xl = 90em * 16 = 1440 , so when `leftOpened = false`, the navbar hidden, 
And the `paddingLeft` of the content on the right should be equal to 0,but `paddingLeft` is not change
```jsx
<Navbar hiddenBreakpoint={'xl'}   width={{ sm: 250 }}
      hidden={!leftOpened}
   
       />
```
<img width="1308" alt="image" src="https://github.com/mantinedev/mantine/assets/124666577/2e89a5cb-1b2a-42a9-9a35-bd5890a09c1f">

**Reply by rtivital** [reply]
You are missing navbarOffsetBreakpoint, see example – https://mantine.dev/core/app-shell/#responsive-styles

**Reply by vaynevayne** [reply]
https://stackblitz.com/edit/stackblitz-starters-hhesfs?file=src%2FApp.tsx 
not work , padddingLeft  can't  update

**Comment by BenSilver1337** [comment]
Based on the provided code snippet, it appears you are using an <AppShell> component that includes a Navbar and MyHeader component. You want to hide the Navbar when the right side of the <AppShell> automatically fills the entire width.

To achieve this, you can use CSS media queries to conditionally hide the Navbar when the desired breakpoint is reached. Assuming that your AppShell component renders the Navbar and the rest of the content, you can add the following CSS styles to hide the Navbar at the specified breakpoint (e.g., xl):

jsx

import React, { useState, useEffect } from 'react';
import { AppShell } from 'your-app-shell-library';

const YourComponent = () => {
  const [leftOpened, setLeftOpened] = useState(true);

  useEffect(() => {
    // Add a listener to detect window resizing
    const handleWindowResize = () => {
      const breakpointWidth = 1280; // You can adjust this to your desired breakpoint width
      setLeftOpened(window.innerWidth >= breakpointWidth);
    };

    // Add the event listener when the component mounts
    window.addEventListener('resize', handleWindowResize);

    // Remove the event listener when the component unmounts
    return () => {
      window.removeEventListener('resize', handleWindowResize);
    };
  }, []);

  return (
    <AppShell
      padding="md"
      navbar={
        <Navbar
          authRouters={authRouters}
          hiddenBreakpoint="xl"
          width={{ sm: 250 }}
          hidden={!leftOpened}
          p="md"
        />
      }
      header={<MyHeader authRouters={authRouters} />}
      styles={(theme) => ({
        main: {
          backgroundColor:
            theme.colorScheme === 'dark' ? theme.colors.dark[8] : theme.colors.gray[0],
        },
      })}
    >
      <MyErrorBoundary>
        <Suspense fallback={<Loader />}>
          <Outlet context={{ routers: authRouters }} />
        </Suspense>
      </MyErrorBoundary>
    </AppShell>
  );
};

export default YourComponent;

With this setup, the Navbar will be hidden when the window width is equal to or greater than the specified breakpoint (in this case, xl). You can adjust the breakpoint width as needed to fit your layout requirements.

Keep in mind that this solution assumes you have a state variable leftOpened that determines whether the right side of the AppShell automatically fills the entire width. You should ensure that this state variable is updated correctly in your code based on your layout logic. Additionally, adjust the breakpointWidth to your desired breakpoint value.

**Comment by vaynevayne** [comment]
mantine's breakpoint is already hidden. What I want is that after hiding, the padinnfLeft of the contnet that responds is 0 @BenSilver1337

**Comment by vaynevayne** [comment]
@rtivital  When the navbar is hidden, `--mantine-navbar-width` should be set to 0. Should this be done by the user?

**Comment by vaynevayne** [comment]
@rtivital  need help
